src/core/ui.js

```diff
@@ -79,6 +84,44 @@ document.documentElement.addEventListener("click", () => {
 });
 respecUI.appendChild(menu);
 
+menu.addEventListener("keydown", e => {
+  if (e.key === "Escape" && !menu.hidden) {
+    menu.classList.add("respec-hidden");
+    menu.classList.remove("respec-visible");
+    respecPill.setAttribute("aria-expanded", String(menu.hidden));
+    respecPill.focus();
+    menu.hidden = true;
```

Comment:
This code partially duplicates Line 79-83. Probably better to add a function and use it here and there.

Reply:
Yes, thats the problem. may be, I can write a function to toggle class